src/openforms/registrations/contrib/objects_api/tests/test_utils.py

```diff
@@ -0,0 +1,36 @@
+from django.test import TestCase
+
+from ..utils import escape_html_manually
+
+
+class EscapeHTMLTests(TestCase):
+    def test_given_dict_values_are_escaped(self):
+        sample = {
+            "test1": "normal value",
+            "test2": "<script>alert();</script>",
+            "nested1": {
+                "n_test": {"n_test1": {"escape": "<script>alert();</script>"}},
+            },
+        }
+        expected = {
+            "test1": "normal value",
+            "test2": "&lt;script&gt;alert();&lt;/script&gt;",
+            "nested1": {
+                "n_test": {
+                    "n_test1": {"escape": "&lt;script&gt;alert();&lt;/script&gt;"}
+                },
+            },
+        }
+
+        result = escape_html_manually(sample)
+
+        self.assertEqual(expected, result)
+
+    def test_not_type_of_dict_returns_empty_dict(self):
+        samples = [["test"], "test", 6]
```

Comment:
I don't think the list behaviour is correct here - list items can be strings and those should also be escaped. Additionally, a list item may itself be a nested dict with values that can be escaped, so please add those samples.